cmd/ncproxy/ncproxy.go

```diff
@@ -57,6 +59,86 @@ func (s *grpcService) AddNIC(ctx context.Context, req *ncproxygrpc.AddNICRequest
 	return nil, status.Errorf(codes.FailedPrecondition, "No shim registered for namespace `%s`", req.ContainerID)
 }
 
+func (s *grpcService) ModifyNIC(ctx context.Context, req *ncproxygrpc.ModifyNICRequest) (_ *ncproxygrpc.ModifyNICResponse, err error) {
+	ctx, span := trace.StartSpan(ctx, "ModifyNIC")
+	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
+
+	span.AddAttributes(
+		trace.StringAttribute("containerID", req.ContainerID),
+		trace.StringAttribute("endpointName", req.EndpointName),
+		trace.StringAttribute("nicID", req.NicID))
+
+	log.G(ctx).WithField("iov settings", req.IovPolicySettings).Info("ModifyNIC iov settings")
+
+	if req.ContainerID == "" || req.EndpointName == "" || req.NicID == "" || req.IovPolicySettings == nil {
+		return nil, status.Error(codes.InvalidArgument, "received empty field in request")
+	}
+
+	if client, ok := containerIDToShim[req.ContainerID]; ok {
+		caIOVSettings := &computeagent.IovSettings{
```

Comment:
Nit: Can this be constructed directly in the `computeagent.ModifyNICInternalRequest` structure below, doesn't look like this var is used afterwards